aws/s3.tf

```diff
@@ -1,7 +1,8 @@
 resource "aws_s3_bucket" "tiered_storage" {
-  count  = var.tiered_storage_enabled ? 1 : 0
-  bucket = local.tiered_storage_bucket_name
-  tags   = local.instance_tags
+  count         = var.tiered_storage_enabled ? 1 : 0
+  bucket        = local.tiered_storage_bucket_name
+  tags          = local.instance_tags
+  force_destroy = var.allow_force_destroy
```

Comment:
I'm leery of allowing this. I assume it's required here because we need it for unit testing or whatever. Can we do this differently? Exposing something like this to a customer seems dangerous. At very least, the description needs to be more forceful.